gradio/routes.py

```diff
@@ -1165,6 +1167,32 @@ def robots_txt():
             else:
                 return "User-agent: *\nDisallow: "
 
+        @app.get("/monitoring")
+        async def analytics_login():
+            print(
+                f"Monitoring URL: {app.get_blocks().local_url}monitoring/{app.analytics_key}"
+            )
+            return HTMLResponse("See console for monitoring URL.")
+
+        @app.get("/monitoring/{key}")
+        async def analytics_dashboard(key: str):
+            if key == app.analytics_key:
+                analytics_url = f"/monitoring/{app.analytics_key}/dashboard"
+                if not app.analytics_enabled:
+                    from gradio.analytics_dashboard import data
+                    from gradio.analytics_dashboard import demo as dashboard
+
+                    mount_gradio_app(app, dashboard, path=analytics_url)
+                    dashboard._queue.start()
+                    analytics = app.get_blocks()._queue.event_analytics
+                    data["data"] = analytics
+                    app.analytics_enabled = True
+                return RedirectResponse(
+                    url=analytics_url, status_code=status.HTTP_302_FOUND
```

Comment:
fwiw this essentially allows anyone to "start the monitoring" on anyone else's Gradio app

Reply:
Anyways, if we do do this, how about we also offer the ability for users to set a parameter in `launch()` and if they set the parameter, the monitoring URL is immediately printed. I find that pattern more intuitive (and less cumbersome since you have the link immediately available).

Its also easier to describe and publicize -- get an API dashboard with just a single parameter!

I.e.

1. A user can set `enable_monitoring=True` when the launch the app
2. If they forget, they can visible `/enable-monitoring` or something to start (I'm not sure if we need this at all btw)

Reply:
Nope, only starts the dashboard app when visiting the secret monitoring link, can add `enable_monitoring=True`

Reply:
> If I understand correctly, this will start monitoring only after the first time a user visits this endpoint? Is that intentional - why do we want to do that?

Nvm I misunderstood, this is fine

Reply:
Maybe the parameter is still useful as a way to display the URL right there at launch but not a big deal either way